src/System.IO.Ports/src/System/IO/Ports/SerialPort.Win32.cs

```diff
@@ -13,26 +15,111 @@ public static string[] GetPortNames()
         {
             // Hitting the registry for this isn't the only way to get the ports.
             //
-            // WMI: https://msdn.microsoft.com/en-us/library/aa394413.aspx
-            // QueryDosDevice: https://msdn.microsoft.com/en-us/library/windows/desktop/aa365461.aspx
+            // WMI: https://msdn.microsoft.com/en-us/library/aa394413.aspx [But Requires .NetFramework !]
+            // QueryDosDevice: https://msdn.microsoft.com/en-us/library/windows/desktop/aa365461.aspx [Can produces duplicates eg COM4 also listed as DosDeviceName \\USB#VID_1B4F&PID_214F..]
             //
-            // QueryDosDevice involves finding any ports that map to \Device\Serialx (call with null to get all, then iterate to get the actual device name)
+            // QueryDosDevice involves finding any ports that map to \Device\Serialx (call with null to get all, then iterate to get the actual device name) [Typically maps to COMx value]
+            // https://docs.microsoft.com/en-gb/previous-versions/ff546502%28v%3dvs.85%29
+            //
+            // GetPortNames will return serial port device names for both registry registered SerialComm devices as well as DosDevice names from Interop QueryDosDevice()
+            // USB Serial Ports on Win10IoT are not initalised in registry key HKLM\HARDWARE\DEVICEMAP\SERIALCOMM correctly, placing garbage chars in this keys value
+            // So to handle this, we are returning a list of port names from both locations.
+            // ToDo: If duplicates wanted to be removed, enumerate the rgistery and try and match & remove duplicate 
+            //   HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\USB is one location, but matching criteria not clear!
+            //   HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Control\COM Name Arbiter\Devices (empty on Win10IoT)
+
+            HashSet<string> resultPortNames = new HashSet<string>();
+
+            // If running on Windows IoT, search and add the Serial Devices from QueryDosDevice
+            // If this is allowed to happen on Windows devices where port detection happens normally, duplicate Serial Port names are added.
```

Comment:
will all devices be duplicated? if so should we just not call the other API at all when running on ARM? If not is there some easy way to remove duplicates?